components/bio-formats-plugins/src/loci/plugins/out/Exporter.java

```diff
@@ -656,8 +656,14 @@ else if (proc instanceof ShortProcessor) {
             int start = doStack ? 0 : imp.getCurrentSlice() - 1;
             int end = doStack ? size : start + 1;
 
-            boolean littleEndian =
-                    !w.getMetadataRetrieve().getPixelsBinDataBigEndian(0, 0).booleanValue();
+            boolean littleEndian = false;
+            if (w.getMetadataRetrieve().getPixelsBigEndian(0) != null)
+            {
+              littleEndian = !w.getMetadataRetrieve().getPixelsBigEndian(0).booleanValue();
+            }
+            else if (w.getMetadataRetrieve().getPixelsBinDataCount(0) == 0) {
+              littleEndian = !w.getMetadataRetrieve().getPixelsBinDataBigEndian(0, 0).booleanValue();
+            }
```

Comment:
It certainly does get reused a lot and a nicer solution would be preferred. The logic was originally found in the hardcoded getPixelsBinDataBigEndian, I would have preferred to try and and keep these sort of exceptions outside of the codegen if possible, otherwise it would almost be simpler to go back to the hardcoded function. I guess ideally it would seem to make sense to only have a single value of big endian on the Pixels object.